barretenberg/cpp/src/barretenberg/honk/execution_trace/execution_trace_usage_tracker.hpp

```diff
@@ -230,51 +230,22 @@ struct ExecutionTraceUsageTracker {
     static std::vector<std::vector<Range>> construct_ranges_for_equal_content_distribution(
         const std::vector<Range>& union_ranges, const size_t num_threads)
     {
-        // Compute the minimum content per thread (final thread will get the leftovers = total_content % num_threads)
-        size_t total_content = 0;
-        for (const Range& range : union_ranges) {
-            total_content += range.second - range.first;
-        }
-        size_t content_per_thread = total_content / num_threads;
-        if (content_per_thread == 0) { // There are more threads than work
-            // Put all work at the back
-            std::vector<std::vector<Range>> thread_ranges(num_threads);
-            thread_ranges.back() = union_ranges;
-            return thread_ranges;
-        }
-
-        std::vector<std::vector<Range>> thread_ranges = { {} };
-        size_t thread_space_remaining = content_per_thread; // content space remaining in current thread
+        std::vector<std::vector<Range>> thread_ranges(num_threads);
 
         for (const Range& range : union_ranges) {
-
             size_t start_idx = range.first;
-            size_t content_to_distribute = range.second - start_idx;
-            while (content_to_distribute >= thread_space_remaining) {
-                // There's enough content to fill the remaining space in the current thread
-                size_t end_idx = start_idx + thread_space_remaining;
-                thread_ranges.back().push_back(Range{ start_idx, end_idx });
-                start_idx = end_idx;
-                content_to_distribute -= thread_space_remaining;
-                thread_space_remaining = content_per_thread;
-                // Create a list for the next thread.
-                thread_ranges.push_back({});
+            size_t total_content = range.second - start_idx;
```

Comment:
It may make sense to have a minimum content here beyond which we don't distribute the work at all. Probably not critical but there will be real cases where certain ranges are quite small (on the order of tens or hundreds of elements)

Reply:
It seems like when determining the number of threads, we already make sure that the [minimum group per thread is at least 4.](https://github.com/AztecProtocol/aztec-packages/blob/jh/benchmark-example-remote/barretenberg/cpp/src/barretenberg/polynomials/evaluation_domain.cpp#L21-L28) Do we need to do anything about that threshold? Not sure if we'd benefit from not distributing if we're spawning the same number of threads anyway.